File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Copysurprise.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Copysurprise implements ITableOperate {
 public final static String TAB_FILE_DATA = "/copysurprise.txt";
 public static final int ID_NPCID = 0 ;
public static final int ID_NPC_WEIGHT = 1 ;
public static final int ID_RATE = 2 ;
public static final int MAX_RECORD = 3 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_NpcID;
 public String GetNpcID() { return m_NpcID;}

private String m_NpcWeight;
 public String GetNpcWeight() { return m_NpcWeight;}

private int m_Rate;
 public int GetRate() { return m_Rate;}

public boolean LoadTable(HashMap<Integer, Table_Copysurprise> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Copysurprise> _hash = (HashMap<Integer, Table_Copysurprise> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Copysurprise _values = new Table_Copysurprise();
 _values.m_NpcID =  (String)valuesList.get(ID_NPCID);
_values.m_NpcWeight =  (String)valuesList.get(ID_NPC_WEIGHT);
_values.m_Rate =  Integer.parseInt((String)valuesList.get(ID_RATE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Copysurprise.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Copysurprise : ITableOperate
{ private const string TAB_FILE_DATA = "copysurprise.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_NPCID,
ID_NPC_WEIGHT,
ID_RATE,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_NpcID;
 public string NpcID { get{ return m_NpcID;}}

private string m_NpcWeight;
 public string NpcWeight { get{ return m_NpcWeight;}}

private int m_Rate;
 public int Rate { get{ return m_Rate;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Copysurprise _values = new Tab_Copysurprise();
 _values.m_NpcID =  valuesList[(int)_ID.ID_NPCID] as string;
_values.m_NpcWeight =  valuesList[(int)_ID.ID_NPC_WEIGHT] as string;
_values.m_Rate =  Convert.ToInt32(valuesList[(int)_ID.ID_RATE] as string);

 _hash.Add(nKey,_values); }


}
}

